serializers/avro/src/main/java/io/pravega/schemaregistry/serializer/avro/impl/AvroGenericDeserializer.java

```diff
@@ -23,17 +26,17 @@
 import java.io.InputStream;
 import java.util.concurrent.ConcurrentHashMap;
 
-class AvroGenericDeserializer extends AbstractDeserializer<Object> {
+public class AvroGenericDeserializer extends AbstractDeserializer<Object> {
     private final ConcurrentHashMap<SchemaInfo, Schema> knownSchemas;
 
-    AvroGenericDeserializer(String groupId, SchemaRegistryClient client, @Nullable AvroSchema<Object> schema,
+    public AvroGenericDeserializer(String groupId, SchemaRegistryClient client, @Nullable AvroSchema<Object> schema,
                             SerializerConfig.Decoders decoder, EncodingCache encodingCache) {
         super(groupId, client, schema, false, decoder, encodingCache, true);
         this.knownSchemas = new ConcurrentHashMap<>();
     }
 
     @Override
-    protected Object deserialize(InputStream inputStream, SchemaInfo writerSchemaInfo, SchemaInfo readerSchemaInfo) throws IOException {
+    public final Object deserialize(InputStream inputStream, SchemaInfo writerSchemaInfo, SchemaInfo readerSchemaInfo) throws IOException {
```

Comment:
Is `final` necessary?

Reply:
yes, this is a class that provides the actual format specific deserialization method which shouldnt be overridden.